### DISCUSSION TITLE
Conflicting definitions for 'react' in 8.3.0

### DISCUSSION THREAD

**Discussion by leylines** [TerriaJS/terriajs#6793] [Q&A] [answered]
I have tons of errors like:

node_modules/@visx/legend/lib/legends/Size.d.ts:1:23 - error TS4090: Conflicting definitions for 'react' found at '/Users/<me>/Library/Caches/Yarn/v4/.tmp/4d274e011f9d62bf03312b276d7efcdb.b3ea0fb9e1976fc017dd144ef857889981f95cb8.prepare/node_modules/@visx/legend/node_modules/@types/react/ts5.0/index.d.ts' and '/Users/joroth/Library/Caches/Yarn/v4/.tmp/4d274e011f9d62bf03312b276d7efcdb.b3ea0fb9e1976fc017dd144ef857889981f95cb8.prepare/node_modules/@types/react/index.d.ts'. Consider installing a specific version of this library to resolve the conflict.

with Version 8.3.0 during "yarn install", even after "yarn cache clean".

Any idea what is causing this conflict?

Thanks you and regards,

Jörg

**Comment by na9da** [comment, marked as answer]
Hi Jörg,

Thanks for reporting the error.

Could you please check that
1. The terriamap package.json has the [following resolutions](https://github.com/TerriaJS/TerriaMap/blob/76553db53f1e9264e7091cb4e3817727e51e1795/package.json#L37-L41):

```json
  "resolutions": {
    "@types/react": "^17.0.3",
    "@types/lodash": "4.14.182",
    "@types/node": "^18.15.11"
  },
```

2. Your yarn.lock file is in sync with the [terriamap one](https://github.com/TerriaJS/TerriaMap/blob/main/yarn.lock).

After that please do a `yarn install` again to fix the dependencies. 
Please let me know if that fixes things.

**Reply by leylines** [reply]
I had to start with a new fork of terriajs because my current fork had problems, although I don't know why, because a diff with the upstream was quite moderate. With the new fork and the same changes it works as expected.

Thanks a lot for your help.

Jörg
